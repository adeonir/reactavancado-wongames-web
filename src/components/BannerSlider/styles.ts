import { BannerContainer } from 'components/Banner/styles'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const BannerSliderContainer = styled.div`
  ${({ theme }) => css`
    .slick-dots {
      list-style: none;
      display: flex !important;
      align-items: center;
      justify-content: center;
      margin-top: ${theme.spacings.small};

      li {
        background: ${theme.colors.white};
        margin: 0 ${theme.spacings.xxsmall};
        width: 1.2rem;
        height: 1.2rem;
        border-radius: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        cursor: pointer;

        &.slick-active {
          background: ${theme.colors.primary};
        }
      }
      button {
        width: 1.2rem;
        height: 1.2rem;
        opacity: 0;
        cursor: pointer;
      }
    }
    ${media.greaterThan('large')`
      ${BannerContainer} {
        max-width: 104rem;
        margin: 0 auto;
      }
      .slick-dots {
        flex-direction: column;
        height: 100%;
        position: absolute;
        top: 0;
        right: 0;
        margin: 0;

        li {
          margin: ${theme.spacings.xxsmall} 0;
        }
      }
    `}
  `}
`
