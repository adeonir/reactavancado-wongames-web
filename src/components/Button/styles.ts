import styled, { css, DefaultTheme } from 'styled-components'

import { ButtonProps } from '.'

type ContainerProps = {
  hasIcon: boolean
} & Pick<ButtonProps, 'size' | 'fullWidth'>

const modifiers = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xsmall};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
    height: 3rem;

    svg {
      width: ${theme.font.sizes.small};

      & + span {
        margin-left: ${theme.spacings.xxsmall};
      }
    }
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.small};
    height: 4rem;

    svg {
      width: ${theme.font.sizes.medium};
    }
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xlarge};
    height: 5rem;

    svg {
      width: ${theme.font.sizes.large};
    }
  `,
  fullWidth: () => css`
    width: 100%;
  `,
  withIcon: (theme: DefaultTheme) => css`
    svg + span {
      margin-left: ${theme.spacings.xxsmall};
    }
  `,
}

export const ButtonContainer = styled.button<ContainerProps>`
  ${({ theme, size, fullWidth, hasIcon }) => css`
    background-image: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
    color: ${theme.colors.white};
    border-radius: ${theme.border.radius};
    border: 0;
    cursor: pointer;
    position: relative;
    z-index: ${theme.layers.base};
    text-decoration: none;

    display: inline-flex;
    align-items: center;
    justify-content: center;

    &:before {
      background-image: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
      content: '';
      border-radius: inherit;
      display: block;
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      z-index: -1;
      opacity: 0;
      transition: opacity 0.2s ease;
    }

    &:hover {
      &:before {
        opacity: 1;
      }
    }

    ${!!size && modifiers[size](theme)}
    ${!!fullWidth && modifiers.fullWidth}
    ${!!hasIcon && modifiers.withIcon(theme)}
  `};
`
