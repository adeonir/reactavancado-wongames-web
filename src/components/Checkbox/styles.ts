import styled, { css } from 'styled-components'

import { CheckboxProps } from '.'

export const CheckboxContainer = styled.div`
  display: flex;
  align-items: center;
`

export const Input = styled.input`
  ${({ theme }) => css`
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    appearance: none;
    width: 1.8rem;
    height: 1.8rem;
    border-radius: 0.2rem;
    border: 0.2rem solid ${theme.colors.gray};
    position: relative;
    transition: background border ${theme.transition.fast};

    &::before {
      content: '';
      width: 0.6rem;
      height: 0.9rem;
      border: 0.2rem solid ${theme.colors.white};
      border-top: 0;
      border-left: 0;
      position: absolute;
      top: 0.1rem;
      transform: rotate(45deg);
      transition: background border ${theme.transition.fast};
      opacity: 0;
    }

    &:focus {
      box-shadow: 0 0 0.5rem ${theme.colors.primary};
    }

    &:checked {
      background: ${theme.colors.primary};
      border-color: ${theme.colors.primary};

      &::before {
        opacity: 1;
      }
    }
  `}
`

export const Label = styled.label<Pick<CheckboxProps, 'labelColor'>>`
  ${({ theme, labelColor }) => css`
    color: ${theme.colors[labelColor!]};
    cursor: pointer;
    padding-left: ${theme.spacings.xxsmall};
    line-height: 1;
  `}
`
