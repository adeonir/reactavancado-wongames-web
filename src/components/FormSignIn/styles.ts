import styled from 'styled-components'

export const FormSignInContainer = styled.div``

export const ForgotPassword = styled.a``

export const FormLink = styled.div``
