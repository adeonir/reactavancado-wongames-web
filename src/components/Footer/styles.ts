import { HeadingContainer } from 'components/Heading/styles'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const FooterContainer = styled.footer`
  ${HeadingContainer} {
    text-transform: uppercase;
  }
`

export const Content = styled.div`
  ${({ theme }) => css`
    display: grid;
    gap: ${theme.grid.gutter};
    margin-top: ${theme.spacings.medium};
    grid-template-columns: repeat(2, 1fr);

    ${media.greaterThan('medium')`
    grid-template-columns: repeat(4, 1fr);
    `}
  `}
`

export const Column = styled.div`
  ${({ theme }) => css`
    a,
    span {
      display: block;
      color: ${theme.colors.gray};
      margin-bottom: ${theme.spacings.xxsmall};
      font-size: ${theme.font.sizes.small};
      text-decoration: none;
    }

    a:hover {
      text-decoration: underline;
    }
  `}
`

export const Copy = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    margin-top: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.xsmall};
    text-align: center;
  `}
`
