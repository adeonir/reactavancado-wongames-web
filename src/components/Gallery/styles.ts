import { rgba } from 'polished'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import { theme } from 'styles'

export const GalleryContainer = styled.div`
  .slick-prev,
  .slick-next {
    display: block;
    color: ${theme.colors.white};
    cursor: pointer;
    position: absolute;
    top: 50%;
    width: 2.5rem;
    height: 2.5rem;
    padding: 0;
    transform: translate(0, -50%);
  }

  .slick-prev {
    left: -${theme.spacings.xxlarge};
  }

  .slick-next {
    right: -${theme.spacings.xxlarge};
  }

  .slick-prev.slick-disabled,
  .slick-next.slick-disabled {
    visibility: hidden;
  }

  .slick-slide > div {
    margin: 0 ${theme.spacings.xsmall};
    cursor: pointer;
  }

  .slick-list {
    margin: 0 -${theme.spacings.xsmall};
  }

  ${media.lessThan('huge')`
    overflow-x: hidden;
  `}
`

type ModalProps = {
  isOpen: boolean
}

const modifiers = {
  open: () => css`
    opacity: 1;
  `,
  close: () => css`
    opacity: 0;
    pointer-events: none;
  `,
}

export const Modal = styled.div<ModalProps>`
  ${({ isOpen }) => css`
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background: ${rgba(theme.colors.black, 0.5)};
    backdrop-filter: blur(5px);
    display: flex;
    align-items: center;
    justify-content: center;
    index: ${theme.layers.modal};
    transition: opacity ${theme.transition.default};

    ${isOpen && modifiers.open}
    ${!isOpen && modifiers.close}
  `}
`

export const Close = styled.div`
  color: ${theme.colors.white};
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  cursor: pointer;
  padding: ${theme.spacings.xsmall};
  text-align: right;
`
