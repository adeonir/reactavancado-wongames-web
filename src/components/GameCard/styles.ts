import { shade, tint } from 'polished'
import styled, { css, DefaultTheme } from 'styled-components'

import { theme } from 'styles'

type PriceProps = {
  isPromotional?: boolean
}

const modifiers = {
  default: (theme: DefaultTheme) => css`
    border-radius: ${theme.border.radius};
    color: ${theme.colors.white};
    background: ${theme.colors.secondary};
    padding: 0 ${theme.spacings.xxsmall};
    margin-right: calc(${theme.spacings.xxsmall} / 2);
  `,
  promotional: (theme: DefaultTheme) => css`
    color: ${theme.colors.gray};
    margin-right: ${theme.spacings.xsmall};
    text-decoration: line-through;
  `,
}

export const GameCardContainer = styled.article`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: ${theme.colors.white};
  position: relative;
`

export const ImageBox = styled.div`
  width: 100%;
  height: 14rem;
  background: ${shade(0.02, theme.colors.white)};
  background-image: linear-gradient(
    to right,
    ${shade(0.02, theme.colors.white)} 0%,
    ${tint(0.78, theme.colors.secondary)} 20%,
    ${shade(0.02, theme.colors.white)} 40%,
    ${shade(0.02, theme.colors.white)} 100%
  );
  background-size: 80rem 14rem;
  animation: shimmer 1s linear infinite forwards;

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  @keyframes shimmer {
    0% {
      background-position: -40rem 0;
    }
    100% {
      background-position: 40rem 0;
    }
  }
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
  height: 100%;
  margin: ${theme.spacings.xsmall};
`

export const Info = styled.div`
  max-width: calc(100% - 2.5rem);
`

export const Title = styled.h3`
  color: ${theme.colors.black};
  font-size: ${theme.font.sizes.medium};
  line-height: ${theme.font.sizes.medium};
  font-weight: ${theme.font.bold};
`

export const Developer = styled.h4`
  color: ${theme.colors.gray};
  font-size: ${theme.font.sizes.small};
  font-weight: ${theme.font.normal};
`

export const FavButton = styled.div`
  color: ${theme.colors.primary};
  cursor: pointer;
  position: absolute;
  right: 0;
  top: -0.5rem;

  svg {
    width: 2.5rem;
  }
`

export const PriceBox = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-top: ${theme.spacings.xxsmall};
`

export const Price = styled.div<PriceProps>`
  ${({ isPromotional }) => css`
    display: inline-flex;
    align-items: center;
    font-weight: ${theme.font.bold};
    height: 3rem;

    ${!isPromotional && modifiers.default(theme)}
    ${isPromotional && modifiers.promotional(theme)}
  `};
`
