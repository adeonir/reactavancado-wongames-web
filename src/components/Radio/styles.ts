import styled, { css } from 'styled-components'

import { theme } from 'styles'

import { RadioProps } from '.'

export const RadioContainer = styled.div`
  display: flex;
  align-items: center;
`

export const Input = styled.input`
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  width: 1.8rem;
  height: 1.8rem;
  border-radius: 50%;
  border: 0.2rem solid ${theme.colors.gray};
  transition: background border ${theme.transition.fast};
  background: transparent;

  &::before {
    content: '';
    width: 0.8rem;
    height: 0.8rem;
    border-radius: 50%;
    background: ${theme.colors.primary};
    transition: opacity ${theme.transition.fast};
    opacity: 0;
  }

  &:focus {
    box-shadow: 0 0 0.5rem ${theme.colors.primary};
  }

  &:checked {
    border-color: ${theme.colors.primary};

    &::before {
      opacity: 1;
    }
  }
`

export const Label = styled.label<Pick<RadioProps, 'labelColor'>>`
  ${({ labelColor }) => css`
    color: ${theme.colors[labelColor!]};
    cursor: pointer;
    padding-left: ${theme.spacings.xxsmall};
    line-height: 1;
  `}
`
