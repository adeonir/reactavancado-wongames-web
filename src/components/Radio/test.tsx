import { screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { theme } from 'styles'
import { renderWithTheme } from 'utils/tests/helpers'

import { Radio } from '.'

describe('<Radio />', () => {
  it('should render with white label', () => {
    const { container } = renderWithTheme(
      <Radio labelText="Radio" labelFor="check" value="anyValue" />,
    )

    const label = screen.getByText('Radio')
    expect(label).toBeInTheDocument()
    expect(label).toHaveStyle({ color: theme.colors.white })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with black label', () => {
    renderWithTheme(<Radio labelText="Radio" labelColor="black" />)

    const label = screen.getByText('Radio')
    expect(label).toBeInTheDocument()
    expect(label).toHaveStyle({ color: theme.colors.black })
  })

  it('should render without label', () => {
    renderWithTheme(<Radio />)

    expect(screen.queryByLabelText('Radio')).not.toBeInTheDocument()
  })

  it('should dispatch onCheck when label status changes', async () => {
    const onCheck = jest.fn()

    renderWithTheme(
      <Radio
        labelText="Radio"
        labelFor="radio"
        onCheck={onCheck}
        value="anyValue"
      />,
    )

    expect(onCheck).not.toHaveBeenCalled()

    userEvent.click(screen.getByLabelText('Radio'))
    await waitFor(() => {
      expect(onCheck).toHaveBeenCalledTimes(1)
    })
    expect(onCheck).toHaveBeenCalledWith('anyValue')
  })

  it('Should be accessible with tab', () => {
    renderWithTheme(<Radio labelText="Radio" labelFor="radio" />)

    expect(document.body).toHaveFocus()

    userEvent.tab()
    expect(screen.getByLabelText('Radio')).toHaveFocus()
  })
})
