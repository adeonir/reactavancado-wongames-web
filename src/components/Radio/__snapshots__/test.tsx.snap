// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should render with white label 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 1.8rem;
  height: 1.8rem;
  border-radius: 50%;
  border: 0.2rem solid #8f8f8f;
  -webkit-transition: background border 0.1s ease-in-out;
  transition: background border 0.1s ease-in-out;
  background: transparent;
}

.c1::before {
  content: '';
  width: 0.8rem;
  height: 0.8rem;
  border-radius: 50%;
  background: #f231a5;
  -webkit-transition: opacity 0.1s ease-in-out;
  transition: opacity 0.1s ease-in-out;
  opacity: 0;
}

.c1:focus {
  box-shadow: 0 0 0.5rem #f231a5;
}

.c1:checked {
  border-color: #f231a5;
}

.c1:checked::before {
  opacity: 1;
}

.c2 {
  color: #fafafa;
  cursor: pointer;
  padding-left: 0.8rem;
  line-height: 1;
}

<div
  class="c0"
>
  <input
    class="c1"
    id="check"
    type="radio"
    value="anyValue"
  />
  <label
    class="c2"
    for="check"
  >
    Radio
  </label>
</div>
`
